src/main/java/de/cronn/validation_files_diff/JoiningStrategy.java

```diff
@@ -18,14 +19,16 @@ public enum JoiningStrategy {
 		this.delimiter = delimiter;
 	}
 
-	public String getPrefix(PsiElement element) {
+	private String getPrefix(PsiElement element) {
 		if (element instanceof PsiClass || element instanceof PsiMethod) {
 			PsiMember psiMember = (PsiMember) element;
-			StringBuilder prefix = new StringBuilder(Objects.requireNonNull(psiMember.getName()));
+			StringBuilder prefix = new StringBuilder(requireNonNull(psiMember.getName()));
+
 			PsiClass containingClass = psiMember.getContainingClass();
 			while (containingClass != null) {
-				String classPart = containingClass.getName() + (prefix.toString().startsWith(delimiter) ? "" : delimiter);
+				String classPart = containingClass.getName() + delimiter;
```

Comment:
I think we need the checking for starting with delimiter here, as with underscores and tests starting with underscore, only a single underscore should remain and not a second one added.

Reply:
But If a test name does actually start with a underscore or slash I guess this should also appear in the prefix twice to match.